File: openapi-test-scala-server-spring-client-jdk/testapi/model/Currency.scala
package testapi.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

@JsonCreator
/** ISO 4217 currency code */
case class Currency(@JsonValue value: String) extends AnyVal

object Currency {
  /** Converts a String to this type for Spring @PathVariable binding */
  def valueOf(s: String): Currency = new testapi.model.Currency(s)
}

File: testers/openapi/java/spring/testapi/model/Currency.java
package testapi.model;

import com.fasterxml.jackson.annotation.JsonValue;

/** ISO 4217 currency code */
public record Currency(@JsonValue String value) {
  public Currency withValue(String value) {
    return new Currency(value);
  }
  ;

  @Override
  public String toString() {
    return value;
  }
  ;

  /** Converts a String to this type for Spring @PathVariable binding */
  public static Currency valueOf(String s) {
    return new Currency(s);
  }
  ;
}
